.github/workflows/ci.yml

```diff
@@ -177,23 +204,33 @@ jobs:
 
           echo "config_key=$config_key" >> $GITHUB_OUTPUT
 
-      - name: Push updated challenge scores
-        if: github.event_name == 'push'
+      - id: push_challenge_scores
+        name: Push updated challenge scores
         run: |
           score_file="tests/challenges/current_score.json"
-
+          challenge_scores_branch="challenge-scores-${{ github.event.pull_request.number }}"
+          echo "challenge_scores_branch=$challenge_scores_branch" >> $GITHUB_OUTPUT
+          
           if ! git diff --quiet $score_file; then
             git add $score_file
             git commit -m "Update challenge scores"
-            git push origin HEAD:${{ github.ref_name }}
+            if [ "${{ startsWith(github.event_name, 'pull_request') }}" = "true" ]; then
+              git checkout -b $challenge_scores_branch
+              git remote add base_repo \
+                https://github.com/${{ github.event.pull_request.base.repo.owner.login }}/${{ github.event.pull_request.base.repo.name }}.git
+              git push -f base_repo $challenge_scores_branch
```

Comment:
I'm not very enthusiastic about the number of data branches this is going to spawn in the main repo. It's going to make my life harder, and I have to wonder if there isn't a better solution than storing transient metadata in the repo.

Reply:
yeah let's discuss to find the best solution